pkg/terminal/command_test.go

```diff
@@ -1113,6 +1113,49 @@ func TestExamineMemoryCmd(t *testing.T) {
 			t.Fatalf("expected first line: %s", firstLine)
 		}
 	})
+
+	withTestTerminal("examinememory_expr", t, func(term *FakeTerminal) {
+		tests := []struct {
+			Expr string
+			Want int
+		}{
+			{Expr: "myInt", Want: 23},
+			{Expr: "&myInt", Want: 23},
+			// TODO: this fails, probably beause it's printing the value of myIntPtr
+			// rather than the value of the addr it is pointing to.
+			{Expr: "myIntPtr", Want: 23},
```

Comment:
I decided to add this new test to make sure the semantics of `examinememory <expression>` are well defined for all the standard Go types.

One thing I discovered is that my current implementation treats pointers differently depending on whether or not the expression itself is creating the pointer (`&myInt`) or if the variable itself already is a pointer (`myIntPtr`). IMO that's inconsistent and should be changed.

But the question is: How should this work? Should `examinememory &myInt` print the addr the pointer is pointing to, or the addr where the pointer is stored, i.e. the pointer value itself?

I can see both options to make some sense, so I figured I should see if y'all have opinions.

Alternatively we could reduce the scope of this patch to only support `addr + offset` kind of expressions officially, and declare any different kind of expression in `examinememory` as undefined behavior for now.

Reply:
With the way you wrote the documentation you should probably only handle pointers (`val.Kind == reflect.Ptr`). You could also say that untyped integer expressions are also allowed and interpreted as addresses and do that too.

Reply:
"handle pointers" ... ok, but how? Should I print the memory at the addr the pointer is stored, or that the pointer is pointing to. And should `myIntPtr` be treated the same way as `&myInt`?

Reply:
> "handle pointers" ... ok, but how? 

You should check that the type of the expression is a pointer and then use the value of the pointer, so &x and ptrToX would behave in the same way and `examinemem -x x` would not work.

Reply:
Got it. Done!